discojs-node/src/loaders/text.ts

```diff
@@ -1,14 +1,98 @@
-import * as fs from "node:fs/promises";
-import * as readline from "node:readline/promises";
+import createDebug from "debug";
+import { createReadStream } from 'node:fs';
 
-import { Dataset, Text } from "@epfml/discojs";
+import { PreTrainedTokenizer } from '@xenova/transformers';
+import { Dataset, Text, models } from "@epfml/discojs";
 
-export function load(path: string): Dataset<Text> {
+const debug = createDebug("discojs-node:loaders:text");
+
+/**
+ * Returns a Dataset that streams and tokenizes text to yield tokenized sequences
+ * one at a time. Each sequence has size `blockSize` + 1, where the first `blockSize`
+ * tokens are the input and the last token is the label. The following sequence
+ * starts with the last token of the previous sequence (so the previous label is now the 
+ * first input token).
+ * In other words, the dataset yields sequences of size `blockSize` + 1 but with an overlap
+ * of 1 token between each sequence.
```

Comment:
Yes the one-token overlap is to make sure that that we predict the next token of each token in the block. And what we are currently doing is fairly predicting the next token of every single token in the batch (the mess going on in the attention layer actually performs the "slide over" the block via matrix operations)
This video bypasses the tokenization complexity by predicting at the character-level (i.e. the next letter rather than the next token). In any case, the video ends up by implementing the batches [similarly to us](https://github.com/karpathy/ng-video-lecture/blob/52201428ed7b46804849dea0b3ccf0de9df1a5c3/gpt.py#L41) by yielding batches with one-token overlap.
Similarly a more recent video/repository addresses the tokenization and also yields batches with overlap https://github.com/karpathy/build-nanogpt/blob/631f7d6f080effc4342b5e2f2f41879642c76fd1/train_gpt2.py#L184
Is it clearer or more confusing?